tests/libnet/cloudinit/cloudinit.go

```diff
@@ -141,19 +141,19 @@ type CloudInitNetworkData struct {
 
 type CloudInitInterface struct {
 	name           string
-	AcceptRA       *bool                `json:"accept-ra,omitempty"`
-	Addresses      []string             `json:"addresses,omitempty"`
-	DHCP4          *bool                `json:"dhcp4,omitempty"`
-	DHCP6          *bool                `json:"dhcp6,omitempty"`
-	DHCPIdentifier string               `json:"dhcp-identifier,omitempty"` // "duid" or  "mac"
-	Gateway4       string               `json:"gateway4,omitempty"`
-	Gateway6       string               `json:"gateway6,omitempty"`
-	Nameservers    CloudInitNameservers `json:"nameservers,omitempty"`
-	MACAddress     string               `json:"macaddress,omitempty"`
-	Match          CloudInitMatch       `json:"match,omitempty"`
-	MTU            int                  `json:"mtu,omitempty"`
-	Routes         []CloudInitRoute     `json:"routes,omitempty"`
-	SetName        string               `json:"set-name,omitempty"`
+	AcceptRA       *bool                 `json:"accept-ra,omitempty"`
+	Addresses      []string              `json:"addresses,omitempty"`
+	DHCP4          *bool                 `json:"dhcp4,omitempty"`
+	DHCP6          *bool                 `json:"dhcp6,omitempty"`
+	DHCPIdentifier string                `json:"dhcp-identifier,omitempty"` // "duid" or  "mac"
+	Gateway4       string                `json:"gateway4,omitempty"`
+	Gateway6       string                `json:"gateway6,omitempty"`
+	Nameservers    *CloudInitNameservers `json:"nameservers,omitempty"`
```

Comment:
@brianmcarey could you please elaborate on the reason this change is needed?

Reply:
I dug the reason out of old slack chats
> Can you try making this a pointer? I believe it would W/A this issue
https://github.com/kubevirt/kubevirt/blob/main/tests/libnet/cloudinit.go#L147
That field should not even exist on v2 network config. It's a bug
But we can avoid this bug when dropping nameservers (which is empty anyway)

Reply:
Thank you @akalenyu